integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanChargePaymentWithAdvancedPaymentAllocationTest.java

```diff
@@ -130,15 +130,27 @@ public static void setup() {
                     CommonConstants.RESPONSE_RESOURCE_ID);
             Assertions.assertNotNull(financialActivityAccountId);
         } else {
+            boolean existFinancialActivity = false;
             for (HashMap financialActivity : financialActivities) {
                 HashMap financialActivityData = (HashMap) financialActivity.get("financialActivityData");
                 if (financialActivityData.get("id").equals(FinancialActivityAccountsTest.LIABILITY_TRANSFER_FINANCIAL_ACTIVITY_ID)) {
                     HashMap glAccountData = (HashMap) financialActivity.get("glAccountData");
                     liabilityTransferAccount = new Account((Integer) glAccountData.get("id"), Account.AccountType.LIABILITY);
                     financialActivityAccountId = (Integer) financialActivity.get("id");
+                    existFinancialActivity = true;
                     break;
                 }
             }
+            if (!existFinancialActivity) {
```

Comment:
what is this?

Reply:
@josehernandezfintecheandomx hm?

Reply:
> @josehernandezfintecheandomx hm?

@adamsaghy  That IT was failing when there is already Financial Activity defined but no one of these is for Activity 200, so If the list has values but none one is for Activity 200 then It is created and the boolean flag is for controlling this